src/transformers/modeling_deberta.py

```diff
@@ -0,0 +1,1061 @@
+        self.scale = 1
+        self.reuse_mask = True
+
+
+def get_mask(input, local_context):
+    if not isinstance(local_context, DropoutContext):
+        dropout = local_context
+        mask = None
+    else:
+        dropout = local_context.dropout
+        dropout *= local_context.scale
+        mask = local_context.mask if local_context.reuse_mask else None
+
+    if dropout > 0 and mask is None:
+        if version.Version(torch.__version__) >= version.Version("1.2.0a"):
+            mask = (1 - torch.empty_like(input).bernoulli_(1 - dropout)).bool()
+        else:
+            mask = (1 - torch.empty_like(input).bernoulli_(1 - dropout)).byte()
+
+    if isinstance(local_context, DropoutContext):
+        if local_context.mask is None:
+            local_context.mask = mask
+
+    return mask, dropout
+
+
+class XDropout(torch.autograd.Function):
+    """Optimized dropout function to save computation and memory by using mask operation instead of multiplication."""
+
+    @staticmethod
+    def forward(ctx, input, local_ctx):
+        mask, dropout = get_mask(input, local_ctx)
+        ctx.scale = 1.0 / (1 - dropout)
+        if dropout > 0:
+            ctx.save_for_backward(mask)
+            return input.masked_fill(mask, 0) * ctx.scale
+        else:
+            return input
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        if ctx.scale > 1:
+            (mask,) = ctx.saved_tensors
+            return grad_output.masked_fill(mask, 0) * ctx.scale, None
+        else:
+            return grad_output, None
+
+
+class StableDropout(torch.nn.Module):
+    """Optimized dropout module for stabilizing the training
+
+    Args:
+
+        drop_prob (float): the dropout probabilities
+
+    """
+
+    def __init__(self, drop_prob):
+        super().__init__()
+        self.drop_prob = drop_prob
+        self.count = 0
+        self.context_stack = None
+
+    def forward(self, x):
+        """Call the module
+
+        Args:
+            x (:obj:`torch.tensor`): The input tensor to apply dropout
+
+
+        """
+        if self.training and self.drop_prob > 0:
+            return XDropout.apply(x, self.get_context())
+        return x
+
+    def clear_context(self):
+        self.count = 0
+        self.context_stack = None
+
+    def init_context(self, reuse_mask=True, scale=1):
+        if self.context_stack is None:
+            self.context_stack = []
+        self.count = 0
+        for c in self.context_stack:
+            c.reuse_mask = reuse_mask
+            c.scale = scale
+
+    def get_context(self):
+        if self.context_stack is not None:
+            if self.count >= len(self.context_stack):
+                self.context_stack.append(DropoutContext())
+            ctx = self.context_stack[self.count]
+            ctx.dropout = self.drop_prob
+            self.count += 1
+            return ctx
+        else:
+            return self.drop_prob
+
+
+def MaskedLayerNorm(layerNorm, input, mask=None):
```

Comment:
I don't really like this function - it is quite confusing because a layernorm module is passed and because the function is in camel case. I don't think we should have this function at all. 

The first line: `output = layerNorm(input).to(input)` just calls the layernorm module and the lines following it create a mask and multiply the mask to the output. 
The function is used 3 times, 2 times with mask=None (which means all it does is `output = layerNorm(input).to(input)` and 1 time with a mask. I think we should just move this code:

```python
    if mask.dim() != input.dim():
        if mask.dim() == 4:
            mask = mask.squeeze(1).squeeze(1)
        mask = mask.unsqueeze(2)
    mask = mask.to(output.dtype)
```

to the last `MaskedLayerNorm` function call and delete the function as a whole

Reply:
Yes. We can do that to make the code more clear.